### ISSUE TITLE
Windows Phone Silverlight support

### ISSUE THREAD

**Issue by PaulAchess** [molinch/FFImageLoading#108]
Hi, I was missing the windows phone silverlight support so I implemented it.

Would you like a pull request to integrate it? I just have a small problem on the forms test project that I can't seem to solve (I don't know where I did wrong but if I add the Transformation.WinSL reference it just doesn't compile with the message « payload contains two or more files with the same destination path »). Any help on this would be appreciated.

I also never did a pull request before, so I really hope I won't blow up everything.

**Comment by molinch** [comment]
Hi @PaulAchess that's very cool :+1: 

Yes having a PR would definitely help.
Then @daniel-luberda and myself can have a look at it and then tell you what's going on.

**Comment by PaulAchess** [comment]
The best part is that I used your package just because it was the simplest I found to crop images. 

But it was so powerful I thought it was cooler to try to add support for silverlight rather than find another way to crop images.

**Comment by ravensorb** [comment]
+1 for bringing this in to the main library.

**Comment by daniel-luberda** [comment]
@PaulAchess  That's cool :) I'll take a look at https://github.com/molinch/FFImageLoading/pull/109

To make transformations work on Windows and Forms it's necessary to make Windows Transformations Package assembly info the same as Transformations PCL assembly (version, etc).

**Comment by daniel-luberda** [comment]
@PaulAchess If you could switch to linked files it would speed integration a lot.

https://msdn.microsoft.com/en-us/library/windows/apps/jj714082(v=vs.105).aspx

**Comment by PaulAchess** [comment]
Thanks for your quick answer ! I'll try to add more linked files (I have a few), I was mostly blocked by the namespaces indeed.

What's the process, do I cancel this PR? Or should I make another commit / PR?

**Comment by molinch** [comment]
You can make additional commits in the same PR :)

**Comment by PaulAchess** [comment]
Perfect, I'm on to it. Thanks for your help :)

**Comment by daniel-luberda** [comment]
@PaulAchess Cool, it would be much easier to review. Thanks for your collaboration :)

**Comment by PaulAchess** [comment]
Glad I could be of some help for your amazing library :)

**Comment by daniel-luberda** [comment]
FFImageLoading Silverlight support is done.

**Comment by daniel-luberda** [comment]
@PaulAchess @ravensorb Did you try 2.0.3-alpha1 nuget if it's working fine on silverlight? I can't test it right now.

**Comment by PaulAchess** [comment]
Not yet, I was on a tight release since yesterday and I only had the time to test my own version. It's in prod, no issue so far, but I will test the alpha asap.

**Comment by PaulAchess** [comment]
Okay, it works fine!

However, there is a small thing to correct on the nuspec somewhere, `Xamarin.FFImageLoading.Transformations` do not need `FFImageLoading.Windows` but rather `Xamarin.FFImageLoading` ([here](https://github.com/molinch/FFImageLoading/blob/master/source/Xamarin.FFImageLoading.Transformations.nuspec#L49))

Or `Xamarin.FFImageLoading.Forms` need `FFImageLoading.Windows` instead of `Xamarin.FFImageLoading` ([here](https://github.com/molinch/FFImageLoading/blob/master/source/Xamarin.FFImageLoading.Forms.nuspec#L54)), but there is an incoherence somewhere.

My solution did not embed `FFImageLoading.Windows` at all, SL libraries where compiled with `Xamarin.FFImageLoading`, as most cross-platform library do.

Actually I wonder why `Xamarin.FFImageLoading` and `FFImageLoading.Windows` are not a single package? It could be merged into `FFImageLoading`. I think it truly makes sense to merge them: FFImageLoading is just an implementation on windows of FFImageLoading, as much as Xamarin.FFImageLoading is an implementation for Android / iOS.

I would make more sense to have a `FFImageLoading` package that supports Windows (SL, RT, UWP), Xamarin and Android in my opinion.

**Comment by PaulAchess** [comment]
Also the description of the Forms package can be changed to add WinSL support

**Comment by molinch** [comment]
I don't want a single package because I use Xamarin Studio only.
On XS you cannot do any Windows development. This means I wouldn't be able anymore to upload nugets for FFImageLoading (non Forms).

**Comment by PaulAchess** [comment]
Okay, I see the issue.

Would it be possible to use a community version of Visual Studio to build it? I really think having only one package would be a plus for users and maintainability.

Nevertheless the incoherence still remains in the packages.

**Comment by molinch** [comment]
It's not really difficult for users: they take FFImageLoading.Forms package and then all dependencies are resolved automatically.
If they only want Windows stuff without Forms they take the Windows package.
If they only want iOS/Android stuff they take the other package.

Maintenance, for me, is way more complicated if we bring Windows in the same nuget: I'm using a Mac with XS.

**Comment by PaulAchess** [comment]
Not really difficult, more counter-intuitive. Fundamentally, Xamarin.FFImageLoading and FFImageLoading.Windows are the same packages, but both are implementations of the same FFImageLoading.

I completely understand your reasons not to merge them, this was merely a suggestion to simplify things.

With this in mind, regarding the nuspec I think then that the package `FFImageLoading.Forms` should embed `FFImageLoading.Windows` ([here](https://github.com/molinch/FFImageLoading/blob/master/source/Xamarin.FFImageLoading.Forms.nuspec#L54)) for WP8, instead of `Xamarin.FFImageLoading`. I should not need this package. 

Even worse right now if I install `Xamarin.FFImageLoading.Forms` it will not embed the needed libraries.

**Comment by ravensorb** [comment]
Would a Shared Project be easier to work with for this?  Move the files that are the same in to the Shared Project and then reference the Shared Project form both packages?

**Comment by daniel-luberda** [comment]
> However, there is a small thing to correct on the nuspec somewhere,  Xamarin.FFImageLoading.Transformations do not need FFImageLoading.Windows but rather Xamarin.FFImageLoading (here)

> Or Xamarin.FFImageLoading.Forms need FFImageLoading.Windows instead of Xamarin.FFImageLoading (here), but there is an incoherence somewhere.

Could you clarify if everything is working fine? Do you get any unused package installed? We could make a unified package in the future - but we'll stick to 4 nugets for now (Windows, XamariniOS/Android, Forms and Transformations).

**Comment by PaulAchess** [comment]
There are 4 packages:

* `FFImageLoading.Windows`
* `Xamarin.FFImageLoading`
* `Xamarin.FFImageLoading.Forms`
* `Xamarin.FFImageLoading.Transformations`

Right now on WSL:
* `Xamarin.FFImageLoading.Forms` embeds `Xamarin.FFImageLoading`
*  `Xamarin.FFImageLoading.Transformations` embeds `FFImageLoading.Windows`

I need both `Xamarin.FFImageLoading.Forms` and `Xamarin.FFImageLoading.Transformations` in my project, which means all 4 packages are imported and everything works fine.

But if I was only using `Xamarin.FFImageLoading.Forms`, it would embed `Xamarin.FFImageLoading` (useless because it does not contain the WSL libraries) and I would be missing `FFImageLoading.Windows` (not embedded by the nuspec package).

Which will probably fail at runtime.

If you stick to 4 packages (and I completely understand this choice), the fix would be to change [this line in Xamarin.FFImageLoading.Forms.nuspec](https://github.com/molinch/FFImageLoading/blob/master/source/Xamarin.FFImageLoading.Forms.nuspec#L54) and make the dependency towards `FFImageLoading.Windows` instead of `Xamarin.FFImageLoading` (or make only one group target for all windows platforms)

**Comment by daniel-luberda** [comment]
Yes, I understand it now. Thanks for detailed explanation! I will fix nuspec files now :)

**Comment by PaulAchess** [comment]
Could you group it with `"portable-netcore45+wpa81"` ? It's the same dependencies.

**Comment by daniel-luberda** [comment]
I would prefer to keep distinct group for Silverlight, something like that: 
```XML
          <group targetFramework="portable-wp8">
            <dependency id="FFImageLoading.Windows" version="2.0.3" />
          </group>  
```

**Comment by daniel-luberda** [comment]
@PaulAchess Could you take a look at this: https://github.com/molinch/FFImageLoading/commit/ad9f675ef46e2c601db7069268d0356a429c6f88

I hope there are no more issues.

**Comment by PaulAchess** [comment]
That seems correct yes!

**Comment by daniel-luberda** [comment]
@PaulAchess Thanks once again! :)

**Comment by PaulAchess** [comment]
You are welcome :)
